src/rust/en.webtoonxyz/src/parser.rs

```diff
@@ -0,0 +1,257 @@
+		"A-Z" => "alphabet",
+		"New" => "new-manga",
+		"No 18+" => "&adult=0&m_orderby=latest",
+		"Only 18+" => "&adult=1&m_orderby=latest",
+		_ => "",
+	};
+
+	let mut url = format!("{}/webtoons/page/{}/?m_orderby={}", base_url, page, list_query);
+
+	if list_query.contains("adult") {
+		url = format!("{}/page/{}/?s&post_type=wp-manga{}", base_url, page, list_query);
+	}
+
+	let (manga_selector, thumb_selector, summary_selector, title_selector) = if list_query.contains("adult") {
+		(".c-tabs-item__content", ".tab-thumb", ".tab-summary", ".post-title")
+	} else {
+		(".manga", ".item-thumb", ".item-summary", ".font-title")
+	};
+
+	let html = Request::new(url, HttpMethod::Get).header("User-Agent", USER_AGENT).html()?;
+	let manga = parse_manga(&html, manga_selector, thumb_selector, summary_selector, title_selector)?;
+	let has_more = !manga.is_empty();
+
+	Ok(MangaPageResult {
+		manga,
+		has_more,
+	})
+}
+
+pub fn parse_manga_details(base_url: String, manga_id: String) -> Result<Manga> {
+	
+	let url = format!("{}/read/{}", base_url, manga_id);
+	let html = Request::new(&url, HttpMethod::Get).header("User-Agent", USER_AGENT).html()?;
+	
+	let manga = html.select(".container");
+	
+	let id = manga_id;
+	let cover = manga.select(".summary_image > a > img").attr("data-src").read();
+	let title = manga.select(".post-title > h1").text().read();
+
+	let info = manga.select(".post-content_item").array();
+
+	let author = info.get(2).as_node()?.select(".author-content").text().read();
+	let artist = info.get(3).as_node()?.select(".artist-content").text().read();
+	let description = manga.select(".summary__content").text().read();
+
+	let categories = manga.select(".genres-content > a")
+		.array()
+		.map(|x| {
+			x.as_node().expect("").text().read()
+		})
+		.collect::<Vec<String>>();
+
+	let mut nsfw_content = MangaContentRating::Safe;
+	if manga.select(".post-title").text().read().contains("18+") {
+		nsfw_content = MangaContentRating::Nsfw;
+	}
+
+	let status_str = info.get(7).as_node()?.select(".summary-content").text().read();
+	let status = if status_str.contains("OnGoing") {
+		MangaStatus::Ongoing
+	} else if status_str.contains("Completed") {
+		MangaStatus::Completed
+	} else {
+		MangaStatus::Unknown
+	};
+
+	Ok(Manga {
+		id,
+		cover,
+		title,
+		author,
+		artist,
+		description,
+		url,
+		categories,
+		status,
+		nsfw: nsfw_content,
+		viewer: MangaViewer::Scroll,
+	})
+}
+
+pub fn parse_chapter_list(base_url: String, manga_id: String) -> Result<Vec<Chapter>> {
+
+	let url = format!("{}/read/{}", base_url, manga_id);
+	let html = Request::new(url.clone(), HttpMethod::Get).header("User-Agent", USER_AGENT).html()?;
+	
+	let mut all_chapters: Vec<Chapter> = Vec::new();
+	
+	for chapter in html.select(".wp-manga-chapter").array() {
+		let chapter = chapter.as_node()?;
+
+		let url = chapter.select("a").attr("href").read();
+		
+		let id = url.split('/').collect::<Vec<&str>>();
+		let id = String::from(id[5]);
```

Comment:
I don't think you need to collect this into a vec before selecting a single index, but I can't quite recall how to do it. maybe with skip or something